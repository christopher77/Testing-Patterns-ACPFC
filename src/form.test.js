import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Form from "./form";

test("The form use setter sended", () => {
  const saveName = jest.fn();
  const { getByLabelText, asFragment } = render(
    <Form label={"label"} placeholder={"placeholder"} saveName={saveName} />
  );
  const input = getByLabelText("label");
  fireEvent.change(input, {
    target: {
      value: "Some name"
    }
  });
  fireEvent.submit(input);
  expect(saveName).toHaveBeenCalledWith("Some name");
  expect(asFragment()).toMatchSnapshot();
});
