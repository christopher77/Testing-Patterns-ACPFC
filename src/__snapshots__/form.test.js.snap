// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The form use setter sended 1`] = `
<DocumentFragment>
  <div
    class="css-11iycnh-Form"
  >
    <form
      class="css-ua5s0o-Form"
    >
      <label
        class="css-c1s87k-Form"
        for="name"
      >
        label
      </label>
      <input
        autocomplete="off"
        class="css-bth6bn-Form"
        id="name"
        name="name"
        placeholder="placeholder"
        required=""
        value=""
      />
      <button
        class="css-1wg226x-Form"
        type="submit"
      >
        Next
      </button>
    </form>
  </div>
</DocumentFragment>
`;
